src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Unix.cs

```diff
@@ -49,6 +51,13 @@ private bool TryConnect(int timeout, CancellationToken cancellationToken)
                 }
             }
 
+            if (!ValidateRemotePipeUser(clientHandle, out Exception exception))
```

Comment:
Seems to me it would be better for ValidateRemotePipeUser to just throw if it needs to? That would eliminate both the return code and the out parameter. It is odd to capture the exception for someone else to throw it. The cleanup you are doing below is exactly what try/finally is for.

Reply:
I want to confirm: my expectation is that the client connecting to a proper chmod'd socket will fail before reaching ValidateRemotePipeUser, is that correct?